src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -657,7 +657,7 @@ trait Contexts { self: Analyzer =>
 
     def enclosingContextChain: List[Context] = this :: outer.enclosingContextChain
 
-    private def treeTruncated       = tree.toString.replaceAll("\\s+", " ").lines.mkString("\\n").take(70)
+    private def treeTruncated       = tree.toString.replaceAll("\\s+", " ").linesIterator.toList.mkString("\\n").take(70)
```

Comment:
Do we need the `toList` here? Seems like it would cause unnecessary allocations. (not like this is a hot method or anything)

Reply:
No, it wasn't necessary. In any case, since moving from `lines` to `linesIterator` is interesting in itself, I am doing it in another PR. #8333